### ISSUE TITLE
request annotation category

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#82]
it might be nice for the user to be able to specify an annotation category to focus on (e.g. "car"), e.g. if a person has an interest in labelled data for a specific application domain ;

Another way to acheive this might be to allow annotating from images chosen in 'explore dataset' (scroll through, find an image of interest, then click on it to annotate it) - that would be more similar to the LabelMe workflow . Maybe that's easier as you wouldn't need to track an extra state in the UI?

**Comment by bbernhard** [comment]
> it might be nice for the user to be able to specify an annotation category to focus on (e.g. "car"), e.g. if a person has an interest in data for a specific application ;

It's almost there - at least in the API. Unfortunately, it's not yet represented in the UI. 

I guess it's because I wasn't sure where to store the selected label preferences. There would be the possibility to store them on the server side bound to a user account (it requires to be signed up), or on the client side (the downside is, that one would lose the settings when using a different browser/device). 

> Another way to acheive this might be to allow chosing an image to annotate from your 'explore dataset' page (i.e. find an image of interest, then click to open it in annotation) - that would be more similar to the LabelMe workflow . Maybe that's easier as you wouldn't need to track an extra state in the UI?

I think a general setting for the "prefered labels" would be nice in any case. But your idea is a nice addition too - I think it could be an interesting way to annotate images. 

> Basically what is happening with me is I upload a batch of street scenes, but if I want to actually annotate them in this platform, I have to skip through pizzas and dogs :)

What about binding the uploaded images to a user account? I could see the following workflow: 

* there will be an option in the users profile page to create one (or more) API tokens. 
* the API token can then be passed along the REST request to glue the uploaded picture to your account
* in case you lose your API token you can revoke the token 

Currently, I am unlocking every picture manually. If there are a lot of pictures waiting in the queue, it takes a while for me to unlock them all. As long as they are locked, you can't annotate/validate them, as they are not showing up. When binding the image to an account it would be possible to weaken that restriction a bit, so that the user that has uploaded the picture can annotate/validate pictures even if they are still in the locked state.

**Comment by dobkeratops** [comment]
> It's almost there - at least in the API. Unfortunately, it's not yet represented in the UI.

does sound like user accounts would smooth it out (see what users focus on)

> Currently, I am unlocking every picture manually. If there are a lot of pictures waiting in the queue, it takes a while for me to unlock them all.

ok that makes sense.. I had assumed it was more about random selection and having a large database (e.g. my contributions would be a small fraction compared to the whole , hence low probability of being selected)

**Comment by dobkeratops** [comment]
>> It's almost there - at least in the API. Unfortunately, it's not yet represented in the UI.

ah, does that mean there's a parameter we could append to the URL? that's enough for me

**Comment by bbernhard** [comment]
> ah, does that mean there's a parameter we could append to the URL? that's enough for me

yes, but unfortunately it's only working in the annotation mode at the moment. But I'll add it to the validation mode as well. 

If you want to try it out: 
https://imagemonkey.io/annotate?label_id=14

Unfortunately, the label id isn't yet returned via API call yet. The reason for that is, that I wasn't sure if I should stick with the database's id as unique identifer or if it's better to introduce a UUID for that purpose. That's why it's a bit hidden ;)

Here are a few label ids: 
car: 14
person: 71
tree: 15
dog: 7

In case you need a full list of label ids, just ping me - I can query the database and create a dump of the labels table for you. 

I think I'll keep the database id as label identifier for now - I can always refactor that later, if needed. I think I should also be able to add the same mechanism to the "Validate Labels" screen. Is there some other screen where you would like to have the ability to filter for labels? (Validate Annotations?)

**Comment by dobkeratops** [comment]
ok those are the main ones to get on with, thats great!
